httpclient-processor/src/main/java/org/springframework/cloud/stream/module/httpclient/HttpClientProcessorConfiguration.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.httpclient;
+
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.config.SpelExpressionConverterConfiguration;
+import org.springframework.cloud.stream.messaging.Processor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * @author Waldemar Hummer
+ * @author Mark Fisher
+ */
+@Configuration
+@EnableBinding(Processor.class)
+@Import(SpelExpressionConverterConfiguration.class)
+@EnableConfigurationProperties(HttpClientProcessorProperties.class)
+public class HttpClientProcessorConfiguration {
+
+	@Bean
+	public RestTemplate restTemplate() {
```

Comment:
The `RestTemplate` is Spring-"free" class. Don't see reason to overhead context with redundant bean.
From other side it would be better to allow people to configure this object. E.g. `ClientHttpRequestFactory`.
We with @garyrussell recently investigated the customer request when he want in addition to the `error-code` from server accepts its custom message as well. So, we found that only `HttpComponentsClientHttpRequestFactory` helps there. And don't forget about `HttpMessageConverter` feature. Plus `ClientHttpRequestInterceptor` and of course, SSL is the most popular.

So, I'd like to hear if you have some solution to inject custom `@Configuration` to an existing module. E.g. with some marker annotation to scan classpath for additional beans. 
With that I can agree on this `RestTemplate` bean, but with `@ConditionalOnMissingBean`, when with the custom `@Configuration` end-user can inject into the module any `RestTemplate`, e.g. `OAuth2RestTemplate` and so on.